### PR TITLE
[KYUUBI #5849] Incorrectly parse JDBC URL while variable includes colon

### LINKED ISSUE #5849: [Bug]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

When some configuration information in our jdbc url contains the same string as the host:port for connecting to kyuubi, this configuration information will be incorrectly overridden. For example, if I configure the following connection string:
`jdbc:hive2://localhost:2182/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=kyuubi-cluster?spark.sql.catalog.test_catalog.url=zookeeper://localhost:2182/default/test_catalog`
 kyuubi will interpret it as an incorrect connection string:
`jdbc:hive2://localhost:10009/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=kyuubi-cluster?spark.sql.catalog.test_catalog.url=zookeeper://dummyhost:00000/default/test_catalog`
replacing my catalog url from `zookeeper://localhost:2182/default/test_catalog` to `zookeeper://dummyhost:00000/default/test_catalog`.

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by hameizi**
# :mag: Description
## Issue References 🔗

This pull request fixes #5849

## Describe Your Solution 🔧

When replacing the `host:port` information in the jdbc url, only the first matching string should be replaced.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [x] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [x] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java [hunk 1]
```diff
@@ -208,7 +208,7 @@ public static JdbcConnectionParams extractURLComponents(String uri, Properties i
       uri = uri.replace(urlPrefix, urlPrefix + authorityFromClientJdbcURL);
     }
     connParams.setSuppliedURLAuthority(authorityFromClientJdbcURL);
-    uri = uri.replace(authorityFromClientJdbcURL, dummyAuthorityString);
+    uri = uri.replaceFirst(authorityFromClientJdbcURL, dummyAuthorityString);
```
Comment (cxzl25) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java:211]: Without this modification, UT can still pass, UT may not need `URLEncoder.encode`.

**Comment by pan3793** [comment]
Thanks for your contribution, could you please revise the PR title and description to reflect the issue more specifically? the current one is quite generic which may confuse successors to explore the project history.

**Comment by pan3793** [comment]
BTW, since this part of code is derived from Apache Hive, does Hive have this issue?

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java (hunk 1 above)
Comment (hameizi) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java:211]: done

**Comment by hameizi** [comment]
> Thanks for your contribution, could you please revise the PR title and description to reflect the issue more specifically? the current one is quite generic which may confuse successors to explore the project history.

done

**Comment by hameizi** [comment]
> BTW, since this part of code is derived from Apache Hive, does Hive have this issue?

Yes, hive also have this issue

**Comment by pan3793** [comment]
@hameizi does Hive have a JIRA ticket?

**Comment by hameizi** [comment]
> @hameizi does Hive have a JIRA ticket, or nobody aware of this issue?

No one has reported this issue with Hive

**Review by pan3793 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.8/1.7

**Pull request closed without merging**